exchange.c

```diff
@@ -165,19 +166,27 @@ static int domain_exchange_once(int (*layoutfunc)(int p), ExchangePlan * plan)
         /* now copy the base P; after PI has been updated */
         partBuf[plan->toGoOffset[target].base + toGoPtr[target].base] = P[i];
         toGoPtr[target].base ++;
-
         /* mark the particle for removal. Both secondary and base slots will be marked. */
         slots_mark_garbage(i);
     }
 
     ta_free(toGoPtr);
     walltime_measure("/Domain/exchange/makebuf");
 
-    /* now remove the garbage particles because they have already been copied.
-     * eventually we want to fill in the garbage gap or defer the gc, because it breaks the tree.
-     * invariance . */
+    /*Find which slots to gc*/
+    int compact[6] = {0};
+    for(ptype = 0; ptype < 6; ptype++) {
```

Comment:
Shall we make this a function, e.g. part of the plan?

Reply:
Done